pulsar-client-cpp/python/pulsar/functions/context.py

```diff
@@ -126,7 +126,8 @@ def record_metric(self, metric_name, metric_value):
   @abstractmethod
   def publish(self, topic_name, message, serde_class_name="serde.IdentitySerDe", properties=None, compression_type=None, callback=None):
     """Publishes message to topic_name by first serializing the message using serde_class_name serde
-    The message will have properties specified if any"""
+    The message will have properties specified if any
```

Comment:
Not sure we want to always do it this way. For example, what if we have multiple input topics with different keys, etc. 
Would it be better to explicitly have an extra parameter?(Both for java and py)

Reply:
I think that should be the default behavior if the user doesn't explicitly specify the key since that is the behavior for objects returned from the function. We can add an additional method in which the user can explicitly specify a key